packages/react-optimistic/src/useLiveQuery.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useMemo } from "react"
+import { useStore } from "@tanstack/react-store"
+import { compileQuery, queryBuilder } from "@tanstack/optimistic"
+import type {
+  Context,
+  InitialQueryBuilder,
+  QueryBuilder,
+  ResultsFromContext,
+  Schema,
+} from "@tanstack/optimistic"
+
+export function useLiveQuery<
+  TResultContext extends Context<Schema> = Context<Schema>,
+>(
+  queryFn: (
+    q: InitialQueryBuilder<Context<Schema>>
+  ) => QueryBuilder<TResultContext>,
+  deps: Array<unknown> = []
+): Map<string, ResultsFromContext<TResultContext>> {
+  const compiledQuery = useMemo(() => {
```

Comment:
really love how slim the React integration is. This will be super fast to port to other frameworks.